cypress-io/cypress
# feat: Implement testing type switch promos

- Closes #26448 
- Closes https://github.com/cypress-io/cypress/issues/26837
- Closes https://github.com/cypress-io/cypress/issues/26836

### Additional details
A new testing type switcher is being added to the Specs Explorer. It has the following behaviors:

### If other testing type is configured
* no "question mark" icon is displayed
* Clicking the other testing type immediately switches the active testing type (closes and reopens browser).

### If other testing type is not configured
* A question mark icon is displayed
* Clicking the other testing type hides the specs list and shows a Promo with informational content about that testing type.
* Promo will:
  - Record an anon event on display
  - Provide a "Quick setup" button that closes browser and sends user to setup flow for other testing type. Clicking button records a `machine-collect` event
  - Provides docs and framework links with appropriate UTM params

[Figma](https://www.figma.com/proto/J6uGNpRC2ct4wIq5j6JzBj/Specs-Page-v1.1-%40latest?page-id=2%3A126&type=design&node-id=579-49216&viewport=355%2C230%2C0.16&scaling=scale-down&starting-point-node-id=579%3A49214&show-proto-sidebar=1)

_Note:_ The following discrepancies with the Figma have been surfaced with design and approved:
1. Different style of Question Mark icon in switcher
2. Switcher is shorter (complies with DS styling)
3. No "skeleton" rows are displayed behind promo
4. There's some extra height from the SpecsList virtualist which causes the promo to center a bit low - going to see how Adam's changes to SpecsList styling impact this

### Steps to test
1. Create a project with CT and E2E set up
5. Open project using this branch, observe switcher displays with no question mark. Clicking each testing type should move Cypress into target testing type.
6. Close project. Edit config file and comment out "component" entry. Reopen project into E2E
7. Switcher should display question mark on CT mode. Click CT button and observe CT promo is displayed. Verify links include appropriate UTM params. Click Quick Setup and verify user is sent to launchpad setup flow for CT.
8. Close project. Edit config file - uncomment "component" and comment out "e2e" config. Reopen project into CT.
9. Switcher should display question mark on E2E mode. Click E2E button and observe E2E promo is displayed. Verify links include appropriate UTM params. Click Quick Setup and verify user is sent to E2E mode.

### How has the user experience changed?

**Both testing types configured**
![Screenshot 2023-05-31 at 1 26 09 PM](https://github.com/cypress-io/cypress/assets/11482842/652f95e0-4a2b-42f0-93f4-386c5c39bcf2)

**CT not configured**
![Screenshot 2023-05-31 at 1 26 56 PM](https://github.com/cypress-io/cypress/assets/11482842/9f6f0e7b-be71-4c96-bdf1-d4913c37db1f)

**E2E not configured**
![Screenshot 2023-05-31 at 1 27 53 PM](https://github.com/cypress-io/cypress/assets/11482842/4dd21b23-b1d8-40ee-b045-f305554de04b)

### PR Tasks
